### PR TITLE
Refactor `dss` commands exit codes and logging

### LINKED ISSUE #78: Add non zero exit code status for errors in dss
### Why it needs to get done

It is a good practice to add non zero exit codes when the CLI fails. These are some [recommendations](https://stackoverflow.com/questions/1101957/are-there-any-standard-exit-status-codes-in-linux). 

### What needs to get done

Add exit codes for errors in DSS

### When is the task considered done

DSS returns with exit code on failure.

### PR DESCRIPTION
**Pull request by NohaIhab**
closes #78 

**Note: Integration tests in the CI are failing due to https://github.com/microsoft/linux-package-repositories/issues/130, but they can be run locally with `tox -e integration`**

## Summary
Refactors the following `dss` commands:
* `create`
* `initialize`
* `logs`
* `remove`
* `status`
* `stop`

With the following changes:
1. use `logger.debug` to store the error with traceback
2. exit with `1` exit code in failure cases
3. raise `RuntimeError` in failure cases inside the module function
4. handle `RuntimeError` in main without logging anything else, just by exiting with status code 1 (logging was done in the module function)
5. handle any other `Exception` in main by providing descriptive error message and use debug to store error and traceback and exit with status code 1.

## Notice on `dss initialize`
This PR refactors `initialize` command in main to catch `RuntimeError` even tho it is not currently raised in the `initialize.py` module. This is because we expect to raise this type of error once we refactor the command, refactoring is being tracked in #68.

### CODE REVIEW

File: src/dss/logs.py [hunk 1]
```diff
@@ -46,15 +49,16 @@ def get_logs(parts: str, name: str, lightkube_client: Client) -> None:
                     namespace=DSS_NAMESPACE,
                     labels=mlflow_deployment.spec.selector.matchLabels,
                 )
-            except ApiError:
+            except ApiError as e:
+                logger.debug(f"Failed to retrieve logs for MLflow: {e}")
```
Comment (misohu) [inline src/dss/logs.py:53]: add the `, exc_info=True`

File: src/dss/main.py [hunk 2]
```diff
@@ -130,10 +148,13 @@ def logs_command(kubeconfig: str, notebook_name: str, print_all: bool, mlflow: b
...
-    kubeconfig = get_default_kubeconfig(kubeconfig)
-    lightkube_client = get_lightkube_client(kubeconfig)
+    try:
+        kubeconfig = get_default_kubeconfig(kubeconfig)
+        lightkube_client = get_lightkube_client(kubeconfig)
 
-    get_status(lightkube_client)
+        get_status(lightkube_client)
+    except RuntimeError:
+        click.get_current_context().exit(1)
```
Comment (misohu) [inline src/dss/main.py:157]: missing except for `Exception` for status command

File: src/dss/main.py
Comment (misohu) [inline src/dss/main.py:1]: For `initialize` command missing except for `RuntimeError`

For `status` command missing except for `Exception`

**Review by misohu — COMMENTED** [review]
Small comments

File: src/dss/main.py
Comment (NohaIhab) [reply src/dss/main.py:1]: I did the change and added a note in the description about this as we discussed

**Review by misohu — APPROVED** [review]
Good job @NohaIhab

**Pull request merged**